loopy/__init__.py

```diff
@@ -487,6 +487,48 @@ def make_copy_kernel(new_dim_tags, old_dim_tags=None):
 # }}}
 
 
+# {{{ einsum
+
+def make_einsum(spec, arg_names, knl_creation_kwargs=None):
+    arg_spec, out_spec = spec.split("->")
+    arg_specs = arg_spec.split(",")
+
+    out_indices = set(out_spec)
+
+    all_indices = set(
+        idx
+        for argsp in arg_specs
+        for idx in argsp) | out_indices
+
+    sum_indices = all_indices - out_indices
+
+    from pymbolic import var
+    lhs = var("out")[tuple(var(i) for i in out_spec)]
+
+    rhs = 1
+    for i, argsp in enumerate(arg_specs):
+        rhs = rhs * var(arg_names[i])[tuple(var(i) for i in argsp)]
```

Comment:
```suggestion
    for arg_name, argsp in zip(arg_names, arg_specs):
        rhs = rhs * var(arg_name)[tuple(var(i) for i in argsp)]
```
multiple uses of "i" was making the code hard to read.

Reply:
Thanks!